NewHaven/src/Resources.cpp

```diff
@@ -48,9 +48,13 @@ HarvestTile::~HarvestTile()
 
 HarvestTile& HarvestTile::operator=(const HarvestTile& tile)
 {
-	tileContent = tile.getTileContent();
+	if (this == &tile)						//Checking for self-assignment
+		return *this;
+
+	delete[] tileContent;					//Delete old data that will be discarded
+	tileContent = tile.getTileContent();	//getTileContent returns an array copy
```

Comment:
Did you test this to make sure it works?

I would suggest doing something like this


```
   for(int i=0; i < 4; i++){
        tileContent[i] = tile.tileContent[i];
    }
```

I have had issues with assigning an array pointer directly where only the head actually got copied and the rest of the array ended up with corrupted and irrelevant data. This may have been caused by an underlying issue, but I wouldnt rely on the copy constructor of a primitive to initiate in your operator override. 

A more standard approach is to using the standard copy library for c++

http://www.cplusplus.com/reference/algorithm/copy/

Reply:
getTileContent() is literally that loop you're talking about. Don't worry about it, the way the array is copied and handled is flawless. There's no way the array is invalid, it was initialized to the heap and only the head is returned.

Reply:
okay